sdk/core/azure-core/src/main/java/com/azure/android/core/http/options/AzureClientOptions.java

```diff
@@ -0,0 +1,76 @@
+package com.azure.android.core.http.options;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.azure.android.core.http.ServiceClient;
+import com.azure.android.core.util.logging.ClientLogger;
+
+public class AzureClientOptions {
+    @NonNull
+    private String apiVersion;
+    @Nullable
+    private ClientLogger logger;
+    @NonNull
+    private TelemetryOptions telemetryOptions;
+    @NonNull
+    private TransportOptions transportOptions;
+
+    /**
+     * Options for configuring a {@link ServiceClient}.
+     *
+     * @param apiVersion       The API version of the service to invoke.
+     * @param logger           The {@link ClientLogger} to be used by the {@link ServiceClient}.
+     * @param telemetryOptions Options for configuring telemetry sent by the {@link ServiceClient}.
+     * @param transportOptions Options for configuring calls sent by the {@link ServiceClient}.
+     */
+    public AzureClientOptions(@NonNull String apiVersion,
+                              @Nullable ClientLogger logger,
```

Comment:
Should we set this to ClientLogger.NONE (or whatever the equivalent is in Java for the do-nothing logger)? Otherwise I worry there are so many places that we log, that it's too easy to forget to check `logger != null` everywhere.

Reply:
We actually have a [default logger](https://github.com/Azure/azure-sdk-for-android/blob/master/sdk/core/azure-core/src/main/java/com/azure/android/core/util/logging/AndroidClientLogger.java) that we can get using [`ClientLogger.getDefault()`](https://github.com/Azure/azure-sdk-for-android/blob/a9c0645976c17c20753d01f8bb3c4dc95c3f3015/sdk/core/azure-core/src/main/java/com/azure/android/core/util/logging/ClientLogger.java#L18). Is it preferred to use the default one or should we provide the option of having a no-op logger as well?

Reply:
Good question, I'll see what we did in iOS. I would -think- that the default is that logging should be on at WARN level (which should already be the default log level).

Reply:
It appears that the default level for both iOS and Android is INFO right now.

Reply:
Ah okay, we will probably want to adjust that at some point.

Checked iOS and we use the default logger if none is provided, and apply the client name as the tag (e.g. "StorageBlobClient").